### PR TITLE
Rework test selection

### PR DESCRIPTION
**Pull request by staticfloat**
This builds two bash arrays, `TESTS_TO_RUN` and `TESTS_TO_SKIP`, which can be added to more flexibly than just building a monolithic `TESTS` string variable.  We then adjust these variables within our various OS or ARCH-specific conditionals, finally building `TESTS` at the end.

This also adjusts win32 to skip the Profile tests, as they are extremely slow because our stack walker is slow on win32, and we're not going to fix that any time soon.

### CODE REVIEW

**Review by DilumAluthge — CHANGES_REQUESTED** [review]
Hmmmm. I personally find this harder to read, i.e. it's hard for me to figure out at a quick glance what tests will be run.

**Comment by staticfloat** [comment]
I think it's much cleaner to just keep track of which tests to run and which to skip, otherwise you have to do things like [this](https://github.com/JuliaCI/julia-buildkite/blob/main/utilities/test_julia.sh#L84) where the conditional is supposed to be related to `Downloads`, but we need to stick `Pkg` in there because we're overwriting what was [previously set on line 78](https://github.com/JuliaCI/julia-buildkite/blob/main/utilities/test_julia.sh#L78).  I feel the "overwriting" behavior is much more confusion.

**Comment by DilumAluthge** [comment]
Yeah, I agree that the old system is also hard to follow.

Let me mull this over a bit more and see if I can come up with something different?

**Review by DilumAluthge — APPROVED** [review]
I've re-reviewed this. I think this is a improvement over the status quo. I do think there might be some tweaks I want to make in the future, but let's get this merged, because it's better than what we have.
